app/models/groceries_items.rb

```diff
@@ -4,6 +4,7 @@ class GroceriesItems < ActiveRecord::Base
 
 	validates :price, numericality: { greater_than_or_equal_to: 0 }
   validates :quantity, numericality: { greater_than_or_equal_to: 0 }
+  validates_uniqueness_of :grocery_id, scope: :item_id
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.